OsmAnd/src/net/osmand/plus/views/MapTileLayer.java

```diff
@@ -126,7 +160,59 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
 		if (mapTileAdapter != null) {
 			mapTileAdapter.onDraw(canvas, tileBox, drawSettings);
 		}
-		drawTileMap(canvas, tileBox, drawSettings);
+
+		final MapRendererView mapRenderer = view.getMapRenderer();
+		if (mapRenderer != null && !oldRender) {
+			//OsmandSettings st = view.getApplication().getSettings();//*
+			//if (!Algorithms.objectEquals(st.MAP_UNDERLAY.get(), cachedUnderlay)) {
+			if (is_set_provider == false) {
+				is_set_provider = true;
```

Comment:
is_set_provider - only 1 time variable? it's going to be changed 1 time in the application and then it's impossible to delete / add /change layer ??? (looks completely wrong)